cardano-c/src/types.rs

```diff
@@ -82,6 +83,44 @@ impl CardanoTransactionErrorCode {
     }
 }
 
+impl From<txbuild::Error> for CardanoTransactionErrorCode {
+    fn from(err: txbuild::Error) -> Self {
+        match err {
+            txbuild::Error::TxInvalidNoInput => Self::no_inputs(),
+            txbuild::Error::TxInvalidNoOutput => Self::no_outputs(),
+            txbuild::Error::TxNotEnoughTotalInput => unimplemented!(),
+            txbuild::Error::TxOverLimit(_) => Self::over_limit(),
+            txbuild::Error::TxOutputPolicyNotEnoughCoins(_) => unimplemented!(),
+            txbuild::Error::TxSignaturesExceeded => Self::signatures_exceeded(),
+            txbuild::Error::TxSignaturesMismatch => Self::signature_mismatch(),
+            txbuild::Error::CoinError(_) => Self::coin_out_of_bounds(),
+            txbuild::Error::FeeError(_) => unimplemented!(),
+        }
+    }
+}
+
+#[repr(C)]
+pub struct Balance {
+    sign: i32,
```

Comment:
I would prefer to use an enum with a `repr(C)` and have the value on the C side as a C enum: `{ DIFF_POSITIVE = X, DIFF_NEGATIVE = X+1, DIFF_ZERO = X+2 }`

Reply:
Something like this? https://github.com/input-output-hk/rust-cardano/pull/575/commits/dd65cf6cec7a6cb8581b18dfe448c85d4e5288b0

Reply:
yes, thanks !